iceoryx_utils/source/posix_wrapper/access_control.cpp

```diff
@@ -104,43 +104,45 @@ bool AccessController::addPermissionEntry(const Category f_category,
     {
     case Category::SPECIFIC_USER:
     {
+        bool result{false};
         if (f_name.empty())
         {
             std::cerr << "Error: specific users must have an explicit name." << std::endl;
-            return false;
+            result = false;
         }
 
         auto id = posix::PosixUser::getUserID(f_name);
         if (!id.has_value())
         {
-            return false;
+            result = false;
         }
         else
         {
-            return addPermissionEntry(f_category, f_permission, id.value());
+            result = addPermissionEntry(f_category, f_permission, id.value());
         }
 
-        break;
+        return result;
     }
     case Category::SPECIFIC_GROUP:
     {
+        bool result{false};
         if (f_name.empty())
         {
             std::cerr << "Error: specific groups must have an explicit name." << std::endl;
-            return false;
+            result = false;
```

Comment:
Same here. The early return avoided that in line 135 `getGroupID` with an empty name was called. Now it is being called and can produce undefined behavior when it is being called with an empty `f_name`